scripts/pre_commit_linter.py

```diff
@@ -343,9 +358,51 @@ def _pre_commit_linter():
     print '\n'.join(summary_messages)
     print ''
 
+    linting_status = 0
     if any([message.startswith(_MESSAGE_TYPE_FAILED) for message in
             summary_messages]):
+        linting_status = 1
+
+    # This part is used for detecting bad patterns.
+    print 'Starting Pattern Checks'
+    print '----------------------------------------'
+    total_files_checked = 0
+    total_error_count = 0
+    all_files = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)]
+    for filename in all_files:
+        if len(filename) != 0:
+            with open(filename) as f:
+                content = f.read()
+                total_files_checked += 1
+                if '__author__' in content:
+                    print filename, ' --> ', BAD_PATTERNS['__author__']
```

Comment:
Generally we use string interpolation:

     print '%s --> %s' % (filename, BAD_PATTERNS['__author__']